src/routes/process-user-vocabulary/components/ImageSelectionOverlay.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { ReactElement, useRef } from 'react'
+import { RNCamera } from 'react-native-camera'
+import styled from 'styled-components/native'
+
+import { CircleIconWhite, ImageIcon } from '../../../../assets/images'
+import CameraOverlay from '../../../components/CameraOverlay'
+import PressableOpacity from '../../../components/PressableOpacity'
+
+const GALLERY_ICON_SIZE = 30
+const TAKE_IMAGE_ICON_SIZE = 50
+
+const Camera = styled(RNCamera)`
+  flex: 1;
+  justify-content: flex-end;
+  position: relative;
+  padding: ${props => props.theme.spacings.sm};
+`
+
+const ActionBar = styled.View`
+  flex-direction: row;
+  width: 100%;
+  align-items: center;
+`
+
+const TakeImageButton = styled.Pressable`
+  flex: 2;
+  align-items: center;
+  margin-right: ${GALLERY_ICON_SIZE}px;
+`
+
+const Container = styled.View`
+  flex-direction: row;
+  justify-content: center;
+`
+
+interface ImageSelectionOverlayProps {
+  setVisible: (visible: boolean) => void
+  numberOfImages: number
+  pushImage: (imageUri: string) => void
+}
+
+const ImageSelectionOverlay = ({ setVisible, pushImage, numberOfImages }: ImageSelectionOverlayProps): ReactElement => {
+  const camera = useRef<RNCamera>(null)
+
+  const takeImage = async () => {
+    if (camera.current) {
+      const options = { quality: 0.5, base64: true }
+      const data = await camera.current.takePictureAsync(options)
+      setVisible(false)
+      if (numberOfImages < 3) {
+        pushImage(data.uri)
+      }
+    }
+  }
+
+  return (
+    <CameraOverlay setVisible={setVisible}>
+      <Camera ref={camera} captureAudio={false} testID='camera'>
+        <Container>
+          <ActionBar>
+            {/* TODO LUN-440 implement gallery */}
+            {/* eslint-disable-next-line no-console */}
+            <PressableOpacity onPress={() => console.log('open gallery')}>
+              <ImageIcon width={GALLERY_ICON_SIZE} height={GALLERY_ICON_SIZE} testID='gallery-icon' />
+            </PressableOpacity>
+            <TakeImageButton onPress={takeImage}>
+              <CircleIconWhite width={TAKE_IMAGE_ICON_SIZE} height={TAKE_IMAGE_ICON_SIZE} testID='take-image-icon' />
```

Comment:
The feedback after taking an image is extremely bad. I didn't realise I've taken an image after I already have made three and got an error message. Will this be addressed in another issue?

Reply:
No, this should work nicely within this ticket. As Steffen suggested, i changed the behaviour, so the cameraOverlay is closed after an image was taken and you see the thumbnail of the image as a feedback. Do you think this is a good way to handle this?

Reply:
Much better now. And probably also the solution with the least amount of work. Maybe we can discuss more improvements at a later point, as they'd probably require more work. :)